cli/src/commands/rebase.rs

```diff
@@ -241,60 +241,16 @@ pub(crate) fn cmd_rebase(
             EmptyBehaviour::Keep,
             "clap should forbid `-r --skip-empty`"
         );
-        let target_commits: Vec<_> = workspace_command
-            .parse_union_revsets(ui, &args.revisions)?
-            .evaluate_to_commits()?
-            .try_collect()?; // in reverse topological order
-        if !args.insert_after.is_empty() && !args.insert_before.is_empty() {
-            let after_commits =
-                workspace_command.resolve_some_revsets_default_single(ui, &args.insert_after)?;
-            let before_commits =
-                workspace_command.resolve_some_revsets_default_single(ui, &args.insert_before)?;
-            rebase_revisions_after_before(
-                ui,
-                command.settings(),
-                &mut workspace_command,
-                &after_commits,
-                &before_commits,
-                &target_commits,
-                &rebase_options,
-            )?;
-        } else if !args.insert_after.is_empty() {
-            let after_commits =
-                workspace_command.resolve_some_revsets_default_single(ui, &args.insert_after)?;
-            rebase_revisions_after(
-                ui,
-                command.settings(),
-                &mut workspace_command,
-                &after_commits,
-                &target_commits,
-                &rebase_options,
-            )?;
-        } else if !args.insert_before.is_empty() {
-            let before_commits =
-                workspace_command.resolve_some_revsets_default_single(ui, &args.insert_before)?;
-            rebase_revisions_before(
-                ui,
-                command.settings(),
-                &mut workspace_command,
-                &before_commits,
-                &target_commits,
-                &rebase_options,
-            )?;
-        } else {
-            let new_parents = workspace_command
-                .resolve_some_revsets_default_single(ui, &args.destination)?
-                .into_iter()
-                .collect_vec();
-            rebase_revisions(
-                ui,
-                command.settings(),
-                &mut workspace_command,
-                &new_parents,
-                &target_commits,
-                &rebase_options,
-            )?;
-        }
+        rebase_revisions(
+            ui,
+            command.settings(),
+            &mut workspace_command,
+            &args.revisions,
+            &args.destination,
+            &args.insert_after,
+            &args.insert_before,
```

Comment:
nit: it's probably better to pass `args` or its subset in to `rebase_revisions()` and `compute_rebase_destination()`.

For example, (maybe as a follow up,) `destination`, `insert_after`, and `insert_before` can be extracted to `sturct RebaseDestinationArgs` and flattened into `RebaseArgs`.

Reply:
I added a new commit to address this.